rest/route/degraded_mode.go

```diff
@@ -0,0 +1,51 @@
+package route
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+)
+
+const firingStatus = "firing"
+const evergreenWebhook = "webhook-devprod-evergreen"
+
+func makeSetDegradedMode() gimlet.RouteHandler {
+	return &degradedModeHandler{}
+}
+
+type degradedModeHandler struct {
+	Receiver string `json:"receiver"`
+	Status   string `json:"status"`
+}
+
+func (h *degradedModeHandler) Factory() gimlet.RouteHandler {
+	return &degradedModeHandler{}
+}
+
+func (h *degradedModeHandler) Parse(ctx context.Context, r *http.Request) error {
+	if err := gimlet.GetJSON(r.Body, h); err != nil {
+		return errors.Wrap(err, "problem parsing request")
+	}
+	if h.Status != firingStatus || h.Receiver != evergreenWebhook {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    "alert is in incorrect state to trigger degraded mode",
+		}
+	}
+	return nil
+}
+
+func (h *degradedModeHandler) Run(ctx context.Context) gimlet.Responder {
+	flags, err := evergreen.GetServiceFlags(ctx)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "retrieving service flags"))
+	}
+	flags.DegradedModeDisabled = false
```

Comment:
Is the intended workflow here that it'll enable degraded mode on high CPU, and disabling it is a manual intervention?

Reply:
Yes, [DEVPROD-6083](https://jira.mongodb.org/browse/DEVPROD-6083) will include updating the ops guide on how to respond to it.